File: gameSrc/model/tile.js
"use strict";

var constants = require("./constants");

/**
 *
 *
 * @constructor
 */
exports.Tile = function () {
    this.type = null;
    this.unit = null;
    this.property = null;
    this.visionTurnOwner = 0;
    this.variant = 0;
    this.visionClient = 0;
};

exports.Tile.prototype = {

    /**
     *
     * @returns {boolean}
     */
    isOccupied: function () {
        return this.unit !== null;
    },

    /**
     *
     * @returns {boolean}
     */
    isVisible: function () {
        return this.visionTurnOwner > 0;
    }

};

File: gameSrc/org/wolfTec/cwt/game/gamemodel/model/Tile.java
package org.wolfTec.cwt.game.gamemodel.model;


public class Tile {

  public Unit unit;
  public Property property;
  public TileType type;

  public int visionTurnOwner = 0;
  public int variant = 0;
  public int visionClient = 0;

  /**
   *
   * @return true if the property is occupied, else false
   */
  public boolean isOccupied() {
    return this.unit != null;
  }

  /**
   * 
   * @return returns true when the turn owner can see this tile.
   */
  public boolean isVisible() {
    return this.visionTurnOwner > 0;
  }

  /**
   * 
   * @return returns true when the turn owner can see this tile.
   */
  public boolean isClientVisible() {
    return this.visionClient > 0;
  }
}
